src/megatron/bridge/training/model_load_save.py

```diff
@@ -323,13 +324,31 @@ def load_megatron_model(
         skip_temp_dist_context: If True, skip temporary distributed context setup.
                                If None, automatically skip if distributed is already initialized.
                                Default: None.
+        override_provider: If provided, use this model config instead of loading from checkpoint.
+                              Must be a TransformerConfig or ModelProviderMixin instance. Default: None.
 
     Returns:
         The model instance with loaded weights if return_state_dict is False,
         otherwise returns a dictionary containing the full, unsharded model state_dict.
     """
 
-    model_cfg, mlm_args = load_model_config(checkpoint_path)
+    if override_provider is not None:
+        model_cfg = override_provider
+        mlm_args = None
```

Comment:
this will break loading MLM models if override_provider is passed in, since mlm_args needs to be available. why not always load the model config and then override model_cfg with override_provider? 

@maanug-nv if you have more thoughts here.

previously, this was a 2-step process for users who needed more customization:

1. call `load_model_config` to get back the transformer config. apply any overrides on this
2. call `build_and_load_model` with the updated config to initialize the megatron model

my concern with passing the override is that users could pass a plain TransformerConfig in and breaking the expectation that it's also a model provider for mbridge usage in L248.

could you share an example for where/how `load_megatron_model` is being called now?